dbms/src/Flash/Mpp/MPPTask.cpp

```diff
@@ -352,6 +352,8 @@ void MPPTask::runImpl()
     else
     {
         context->getProcessList().sendCancelToQuery(context->getCurrentQueryId(), context->getClientInfo().current_user, true);
+        if (dag_context)
+            dag_context->cancelAllExchangeReceiver();
```

Comment:
why before L357? is it better to write errors to the turnnel first?

Reply:
Refering to chat history in our `OOT group`, xufei has explained the reason. Receiver needs close first